src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -1110,11 +1110,40 @@ bool entitiesQuery
     }
 
     /* Process query result */
-    while (cursor->more()) {
+    while (cursor->more())
+    {
+        BSONObj                 r    = cursor->next();
+        std::string             err  = r.getStringField("$err");
+        ContextElementResponse* cer  = new ContextElementResponse();
+
+        if (err != "")
+        {
+          //
+          // We can't return the error 'as is', as it may contain forbidden characters.
+          // So, we can just match the error and sent a less descriptive text
+          //
+          const char* invalidPolygon      = "Exterior shell of polygon is invalid";
+          const char* defaultErrorString  = "Internal mongo problem";
+
+          LM_W(("Database Error (%s)", err.c_str()));
+
+          if (strncmp(err.c_str(), invalidPolygon, strlen(invalidPolygon)) == 0)
```

Comment:
Remember using `{... }` even for single-line if clauses.

In addition, not sure about the "not detected at creation time" message appending... On the one side is like saying that the CB should have detected something but it has failed in doing so (what is not correct, as the CB is not designed to detect "exterior shell problems" so far). On the other side "creation" is a bit confusing... as the subject of the creation is not clear (an entity? a polygon -note that the polygon itself is not being created, but the structure to query for a polygon, which is not exactly the same).

Whatever we use at the end, I think is a good idea to create a tech debt issue (if you haven't already do so) to enhance the CB to detect these "exterior shell problems"